### PR TITLE
Impl helm chart

### PR DESCRIPTION
**Pull request by xDarksome**
## Synopsis

There is no Helm chart yet.

## Solution

Implement Helm chart & add its linting to Makefile / ci.

## Checklist

- Created PR:
    - [x] In [draft mode][l:1]
    - [x] Name contains issue reference
    - [x] Has type and `k::` labels applied
- Before [review][l:4]:
    - [x] Documentation is updated (if required)
    - [x] Tests are updated (if required)
    - [x] Changes conform [code style][l:2]
    - [x] [CHANGELOG entry][l:3] is added (if required)
    - [x] FCM (final commit message) is posted or updated
    - [x] [Draft mode][l:1] is removed
- [x] [Review][l:4] is completed and changes are approved
    - [x] FCM (final commit message) is approved
- Before merge:
    - [x] Milestone is set
    - [x] PR's name and description are correct and up-to-date
    - [x] All temporary labels are removed

[l:1]: https://help.github.com/en/articles/about-pull-requests#draft-pull-requests
[l:2]: /CONTRIBUTING.md#code-style
[l:3]: /CHANGELOG.md
[l:4]: https://help.github.com/en/articles/reviewing-changes-in-pull-requests

### CODE REVIEW

**Comment by xDarksome** [comment]
FCM
---
```
Implement Helm chart (#3)
```

File: helm/baza/templates/statefulset.yaml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        kubectl.kubernetes.io/default-container: baza
+      {{- with .Values.statefulset.revision }}
+        revision: {{ . | quote }}
+      {{- end }}
+      {{- with .Values.statefulset.annotations }}
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+    spec:
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
```
Comment (tyranron) [inline helm/baza/templates/statefulset.yaml:43]: No need for `hostNetwork` here.

**Review by tyranron — CHANGES_REQUESTED** [review]

File: helm/baza/templates/statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        {{- else }}
+          {{- toYaml .Values.statefulset.podAntiAffinity | nindent 10 }}
+        {{- end }}
+      {{- end }}
+    {{- end }}
+    {{- with .Values.statefulset.tolerations }}
+      tolerations: {{- toYaml . | nindent 8 }}
+    {{- end }}
+      containers:
+        - name: baza
```
[the remark this answers is not in the document]
Comment (xDarksome) [reply helm/baza/templates/statefulset.yaml:78]: @tyranron I thought it was gonna be a separate thing. In our specific case we actually need Openresty for JWT validation, while someone else might want to use it with Nginx or something else.

[the remark this answers is not in the document]
Comment (tyranron) [reply helm/baza/templates/statefulset.yaml:78]: @xDarksome yah, but deploying it separately will be cumbersome. We still need to provide Nginx container for reading files directly on a separate port. And in our custom deployment we may substitute the image to OpenResty and provide custom config.

**Review by tyranron — APPROVED** [review]

**Pull request merged**